### PR TITLE
Join add cds cleanup

### PR DESCRIPTION
**Pull request by romu42**
#136 triggered this PR, I went through the code. It was fine, but I updated the Variables to make it easier to follow the logic.

### CODE REVIEW

**Review by johanix — COMMENTED** [review]
- "verifys" --> "verifies"
- if you change the variable z to 'zone' (which is fine) then stuff like zone.ZoneType is a bit redundant. Change the .ZoneType field to just .Type perhaps? Also matches with the .Name that we already use.
- 'cdnskeyFrom...' --> 'cdnskeysFrom...' ?
- You've changed "for k, _ := range ..." to just "for k := range...". I'm strongly against this change. Having the unused result marked as unused (via '_') is a signal that we know there is an additional result and it is an active decision not to use it. Removing that means that it is impossible to se whether we're aware of the second result (good) or completely unaware (bad). I think you agree, because you haven't changed this in other places.

Otherwise it looks fine.
-

**Comment by romu42** [comment]
- Changed "verifys" to "verifies
- Opened a issue #199 to refactor the ZoneType, it changes a lot of files so I feel more comfortable having that in a separate PR
- Not sure by what you mean by : `'cdnskeyFrom...' --> 'cdnskeysFrom...' ?`
- In regards to `k,_ := range...`, there is s difference between if you have a blank identifier as the last iteration variable or not. All other uses in the file are ` _ , VAR := range ITERABLE`  

> **from the go 1.17 spec**  If the last iteration variable is the [blank identifier](https://go.dev/doc/go1.17_spec#Blank_identifier), the range clause is equivalent to the same clause without that identifier.

That being said, I put it back for readability.

**Comment by romu42** [comment]
closes #136

File: fsm/join_add_cds.go [hunk 1]
```diff
@@ -124,106 +123,106 @@ func JoinAddCdsAction(z *music.Zone) bool {
...
-			err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
+			err, _ = zone.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
 				signer.Name, err))
 			return false
 		}
 
-		for _, cdsrr := range cdsrrs { // check all CDS RRs from this signer
-			cds, ok := cdsrr.(*dns.CDS)
+		for _, rr := range cdsRRset {
+			cdsRR, ok := rr.(*dns.CDS)
```
Comment (johanix) [inline fsm/join_add_cds.go:190]: Here cdsRR is a *dns.CDS

File: fsm/join_add_cds.go [hunk 2]
```diff
@@ -124,106 +123,106 @@ func JoinAddCdsAction(z *music.Zone) bool {
...
+			if _, exist := cdsFromKSK[cdsRR.KeyTag]; !exist {
+				err, _ = zone.SetStopReason(fmt.Sprintf("CDS RR with keyid=%d published by signer %s should not exist", cdsRR.KeyTag, signer.Name))
 				return false
 			}
 		}
-		for _, revcds := range cdsmap {
-			if _, exist := cdsmap2[revcds.KeyTag]; !exist {
-				err, _ = z.SetStopReason(fmt.Sprintf("CDS RR with keyid=%d should be published by %s, but is not",
-					revcds.KeyTag, signer.Name))
+		for _, cdsRR := range cdsFromKSK {
```
Comment (johanix) [inline fsm/join_add_cds.go:200]: Here cdsRR is a *dns.RR.

I don't think this improves readability.

**Review by johanix — COMMENTED** [review]
Not a fan. Sorry.

We have to much to do to spend time both for you to do rototilling of variable names and for me to do review of it all.

File: fsm/join_add_cds.go (hunk 1 above)
Comment (romu42) [reply fsm/join_add_cds.go:190]: Ok, I see that. Do you mean that it is incorrect in some way? I don't have enough information in your comment with my knowledge level to make a determination if there is something wrong or not. If there is something wrong do you have a suggestion for an improvement?

File: fsm/join_add_cds.go (hunk 2 above)
Comment (romu42) [reply fsm/join_add_cds.go:200]: As far as I can tell it is a `*dns.CDS`

**Comment by romu42** [comment]
The change of the variable names is to make it easier to follow the code, I find it easier now at least. An issue was opened stating that the code was not robust. When reading the code I found it difficult to follow and decided to clean up the variable naming a bit.  If you don't find this helpful or important I am interested to hear any alternatives to a better way to make such changes with out you having to do a review.

File: fsm/join_add_cds.go (hunk 2 above)
Comment (johanix) [reply fsm/join_add_cds.go:200]: You're right, I'm wrong. Sorry.

**Review by johanix — APPROVED** [review]

**Pull request merged**